### PR TITLE
Add Batch insert + On conflict do update

### PR DESCRIPTION
**Pull request by semtexzv**

### CODE REVIEW

File: base/database/batch.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+// nolint: gochecknoglobals
+var bulkNow time.Time
+
+func BulkInsert(db *gorm.DB, objects []interface{}) error {
+	return bulkExec(db, objects)
+}
+
+// BulkInsertChunk will split the objects passed into the passed chunk size. A
+// slice of errors will be returned (if any).
+func BulkInsertChunk(db *gorm.DB, objects []interface{}, chunkSize int) []error {
```
Comment (Josca) [inline base/database/batch.go]: This is never used and tested.

File: base/database/batch.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	if err != nil {
+		return err
+	}
+
+	// No scope and no error means nothing to do
+	if scope == nil {
+		return nil
+	}
+
+	return db.Exec(scope.SQL, scope.SQLVars...).Error
```
Comment (Josca) [inline base/database/batch.go]: This uses `INSERT INTO "test_tables" ("created_at", "deleted_at", "email", "name", "updated_at") VALUES (?, ?, ?, ?, ?), (?, ?, ?, ?, ?), (?, ?, ?, ?, ?) ` in tests. So it adds some extra fields we are not going to use.

File: base/database/dbtest.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package database
+
+import "github.com/jinzhu/gorm"
+
+type TestTable struct {
+	Name  string
+	Email string
+	gorm.Model
```
Comment (Josca) [inline base/database/dbtest.go]: We don't use `gorm.Model` and `AutoMigrate`, so we should use the same style for testing and using.

File: base/database/batch.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// bulkNow holds a global now state and will be used for each records as
```
Comment (Josca) [inline base/database/batch.go]: Is this code copied from anywhere? We should reference it.

File: base/database/batch.go (hunk 1 above)
Comment (semtexzv) [reply base/database/batch.go]: Added a test for this function.

File: base/database/dbtest.go (hunk 3 above)
Comment (semtexzv) [reply base/database/dbtest.go]: Removed `gorm.Model`, but keeping the `AutoMigrate` as a way to simply generate table in the test database, without requiring  inclusion of the test table into the production database or custom SQL.

File: base/database/batch.go (hunk 4 above)
Comment (semtexzv) [reply base/database/batch.go]: Fixed.

File: base/database/batch.go (hunk 2 above)
Comment (semtexzv) [reply base/database/batch.go]: Behavior is the same as normal `gorm.Create`. The fields were added by `gorm.Model`.

**Pull request merged**